static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -238,6 +246,44 @@ function SelectProjectStep({
   );
 }
 
+function usePublicDSN({
+  organization,
+  project,
+}: {
+  organization: Organization | null | undefined;
+  project: Project | null | undefined;
+}) {
+  const api = useApi();
+  const [response, setResponse] = useState<RequestState<string | null>>({
+    type: 'initial',
+  });
+  useEffect(() => {
+    if (!organization || !project) {
+      return;
+    }
+    setResponse(v => ({...v, type: 'loading', data: null}));
+    const request: Promise<ProjectKey[]> = api.requestPromise(
+      `/projects/${organization.slug}/${project.slug}/keys/`
+    );
+
+    request
+      .then(data => {
+        setResponse({
+          type: 'resolved',
+          data: data[0]?.dsn.public ?? null,
+        });
+      })
+      .catch(error =>
+        setResponse({
```

Comment:
This will attempt to set state on an unmounted component. I think you can return api.clear from the hook to cleanup the side effect or manually track mount/unmount.

Reply:
good call